### DISCUSSION TITLE
can I cache the routes in app/router.options?

### DISCUSSION THREAD

**Discussion by decba78b** [nuxt/nuxt#25058] [Questions]
I need to add an alias in all route, so I try to do this
```
function recursiveHandler (routes) {
  routes.forEach((route) => {
    if (route.children)
      recursiveHandler(route.children)
    if (route.alias)
      route.alias = [...route.alias!, route.path + '.html']
  })
}
……
routes: (_routes) => {
  const routes = [..._routes]
  recursiveHandler(routes)
  return routes
},
……
```
alias will be push in `_routes` so it will be expanded to a large size if the pages has been viewed many times.

Cause of this, can I make a cache routes after I generate a new routes list?and why the routes method runs every time when the page views

**Comment by xjccc** [comment]
you can create file as `.html.vue`?

**Reply by decba78b** [reply]
it's one way to resolve this problem, but it will make all the files have to named with `.html`, it's weird.

**Reply by xjccc** [reply]
use hooks 'page:extend'?
